code/datums/gamemodes/gangwar.dm

```diff
@@ -637,31 +952,74 @@ proc/broadcast_to_all_gangs(var/message)
 	item_state = "spraycan"
 	flags = FPRINT | EXTRADELAY | TABLEPASS | CONDUCT
 	w_class = W_CLASS_SMALL
-	var/in_use = 0
-
-	afterattack(target as turf|obj, mob/user as mob)
-		if(!istype(target, /turf) && !istype(target, /obj/decal/cleanable/gangtag))
-			return
-
-		if (!user)
-			return
-
-		if(in_use)
-			boutput(user, SPAN_ALERT("You are already tagging an area!"))
-			return
-
-		var/turf/turftarget = get_turf(target)
-
-		if((turftarget == loc) || (BOUNDS_DIST(src, target) > 0))
-			return
+	var/in_use = FALSE
+	var/empty = FALSE
+
+	/// Checks a tile has no nearby claims from other tags
+	proc/check_tile_unclaimed(turf/target, mob/user)
+		for (var/obj/decal/gangtag/tag in range(GANG_TAG_SIGHT_RANGE,target))
```

Comment:
Doesn't matter tooo much but it'd almost certainly be faster to iterate all gang tags and check range than iterate range and then have to check range again.

Reply:
i forgor how to do this

Reply:
`for_by_tcl` or iterate over `by_type`, pretty sure you've done it somewhere else in here